pkg/asset/store/store.go

```diff
@@ -307,6 +307,9 @@ func (s *storeImpl) load(a asset.Asset, indent string) (*assetState, error) {
 			logrus.Warningf("%sDiscarding the %s that was provided in the target directory because its dependencies are dirty and it needs to be regenerated", indent, a.Name())
 		}
 		source = unfetched
+	// The Metadata should always be regenerated because it may be stale in state file.
+	case a.Name() == "Metadata":
```

Comment:
The asset graph understands general assets that important the interface.

This change couples the impl to a specific asset based on name. This is not great for maintaining the abstractions and separation of concerns

Reply:
Removed this and went with another method.